client/my-sites/site-settings/publishing-tools/post-by-email.tsx

```diff
@@ -0,0 +1,163 @@
+import isRegeneratingJetpackPostByEmail from 'calypso/state/selectors/is-regenerating-jetpack-post-by-email';
+import isSiteAutomatedTransfer from 'calypso/state/selectors/is-site-automated-transfer';
+import { isJetpackSite } from 'calypso/state/sites/selectors';
+import { getSelectedSiteId } from 'calypso/state/ui/selectors';
+
+type PostByEmailSettingProps = {
+	isFormPending: boolean;
+	address?: string;
+};
+
+const noticeConfig = {
+	duration: 4000,
+};
+
+const moduleSlug = 'post-by-email';
+
+export const PostByEmailSetting = ( { isFormPending, address }: PostByEmailSettingProps ) => {
+	const selectedSiteId = useSelector( getSelectedSiteId ) || 0;
+	const siteIsJetpack = useSelector( ( state ) => isJetpackSite( state, selectedSiteId ) );
+	const siteIsAtomic = useSelector( ( state ) => isSiteAutomatedTransfer( state, selectedSiteId ) );
+	const jetpackRegeneratingPostByEmail = useSelector( ( state ) =>
+		isRegeneratingJetpackPostByEmail( state, selectedSiteId )
+	);
+	const jetpackPostByEmailIsActive = !! useSelector( ( state ) =>
+		isJetpackModuleActive( state, selectedSiteId, moduleSlug )
+	);
+	const siteInDevMode = useSelector( ( state ) =>
+		isJetpackSiteInDevelopmentMode( state, selectedSiteId )
+	);
+	const moduleUnavailableInDevMode = useSelector( ( state ) =>
+		isJetpackModuleUnavailableInDevelopmentMode( state, selectedSiteId, moduleSlug )
+	);
+	const moduleUnavailable = siteInDevMode && moduleUnavailableInDevMode;
+	const { data: simpleSitePostByEmailSettings } = useGetPostByEmail( selectedSiteId );
+	const { mutate: simpleSiteSwitchPostByEmail, isPending: isSimpleSitePendingSwitch } =
+		useTogglePostByEmailMutation( selectedSiteId );
+	const { mutate: simpleSiteRegeneratePostByEmail, isPending: isSimpleSitePendingRegenerate } =
+		useRegeneratePostByEmailMutation( selectedSiteId );
+
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+
+	const handleSwitchForSimpleSite = ( checked: boolean ) => {
+		if ( ! selectedSiteId ) {
+			return;
+		}
+
+		simpleSiteSwitchPostByEmail( checked, {
+			onSuccess: () => {
+				dispatch( successNotice( translate( 'Settings saved successfully!' ), noticeConfig ) );
+			},
+			onError: () => {
+				dispatch(
+					errorNotice( translate( 'There was a problem saving your changes.' ), noticeConfig )
+				);
+			},
+		} );
+	};
+
+	const handleRegenerate = () => {
+		if ( ! selectedSiteId ) {
+			return;
+		}
+
+		if ( siteIsJetpack ) {
+			dispatch( jetpackRegeneratePostByEmail( selectedSiteId ) );
+			return;
+		}
+
+		simpleSiteRegeneratePostByEmail( undefined, {
+			onSuccess: () => {
+				dispatch( successNotice( translate( 'Address regenerated successfully!' ), noticeConfig ) );
+			},
+			onError: () => {
+				dispatch(
+					errorNotice( translate( 'There was a problem regenerating the address.' ), noticeConfig )
+				);
+			},
+		} );
+	};
+
+	const email = address && address !== 'regenerate' ? address : '';
```

Comment:
Below we have extra condition and additionallt this variable doesn't make sense to generate for `! siteIsJetpack` website. So let's keep 1 condition and generate `email` in one place;
```suggestion
	let email;
	if (siteIsJetpack) {
		email = address && address !== 'regenerate' ? address : '';
	} else {
		email = simpleSitePostByEmailSettings?.email;
	}
```